File: dis-ingest/src/main/java/io/pivotal/dis/ingest/store/OngoingDisruptionsStore.java
package io.pivotal.dis.ingest.store;

import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public class OngoingDisruptionsStore {
    private String previousDisruptionDigest;

    public Optional<String> getPreviousDisruptionDigest() {
        return Optional.ofNullable(previousDisruptionDigest);
    }

    public void setPreviousDisruptionDigest(String digest) {
        previousDisruptionDigest = digest;
    }
}

File: dis-ingest/src/main/kotlin/io/pivotal/dis/ingest/app/store/OngoingDisruptionsStore.kt
package io.pivotal.dis.ingest.app.store

class OngoingDisruptionsStore {
    private var _previousDisruptionDigest: String? = null

    var previousDisruptionDigest: String?
        get() = _previousDisruptionDigest
        set(value) {
            _previousDisruptionDigest = value
        }
}
